### PR TITLE
upgrade to terraform 0.12

### PR DESCRIPTION
**Pull request by minniux**
Related issue: https://github.com/sweepbright/infrastructure/issues/440

### CODE REVIEW

File: duply-bucket-iamuser/variables.tf [hunk 1]
```diff
@@ -7,3 +8,4 @@ variable "project" {
 variable "environment" {
   default = "production"
 }
+
```
Comment (simonrondelez) [inline duply-bucket-iamuser/variables.tf:14]: can you define the types on these

File: duply-user/variables.tf [hunk 2]
```diff
@@ -8,4 +9,6 @@ variable "project" {
   default = ""
 }
 
-variable "arn" {}
+variable "arn" {
+}
+
```
Comment (simonrondelez) [inline duply-user/variables.tf:18]: same here: define types

File: duply-bucket-iamuser/variables.tf (hunk 1 above)
Comment (minniux) [reply duply-bucket-iamuser/variables.tf:14]: should be done, If I'm not mistaken (looking at other updated modules), we only define types when there's no default or when type is complex, no?

**Review by venturel — APPROVED** [review]

File: duply-bucket-iamuser/variables.tf (hunk 1 above)
Comment (iuriaranda) [reply duply-bucket-iamuser/variables.tf:14]: with the new complex variable types, I think it's a best practice to enforce the type of all variables, even if they have defaults.

**Review by simonrondelez — APPROVED** [review]

**Pull request merged**